docs/docs/ScalaUserGuide/build-bigdl-core.md

```diff
@@ -0,0 +1,125 @@
+## **Build BigDL-core on Different Platforms**
+
+### **Setup Build Environment**
+
+For building BigDL-core, there should have
+
++ JDK 1.7+
++ maven
++ make
++ g++-7
++ Intel Parallel Studio
++ Git.
+
+BigDL-core is a JNI project, `mkl2017-xeon-blas` needs MKL libraries with icc and `bigquant` needs g++-7. We use `maven` + `make` to control the build process where maven for java and make for c/c++ code.
+
+### **Environment on CentOS**
+
+* **Build GCC-7.2**
+
+```bash
+git clone https://github.com/gcc-mirror/gcc.git
+git checkout gcc-7_2_0-release
+./configure --prefix=/opt/gcc-7.2.0 --enable-languages=c,c++ --disable-multilib --disable-nls
+make -j4 && make install
+ln -s /opt/gcc-7.2.0 /opt/gcc
+```
+
+* **Build binutils 2.29**
+
+```bash
+wget https://ftp.gnu.org/gnu/binutils/binutils-2.29.tar.gz
```

Comment:
why build binutils from source code?

Reply:
GCC-7 need the newer binutils. The default one is so old.